arangod/Aql/OptimizerRules.cpp

```diff
@@ -6640,7 +6640,7 @@ void arangodb::aql::inlineSubqueriesRule(Optimizer* opt,
 
         // we're only interested in FOR loops...
         auto listNode = ExecutionNode::castTo<EnumerateListNode*>(current);
-
+        TRI_ASSERT(listNode->getMode() == EnumerateListNode::kEnumerateArray);
```

Comment:
This assertion requires a commit. It only holds true, because the `replaceEntriesWithObjectIteration` is executed after the `inlineSubqueriesRule` and thus this rule never has to deal with enumeration nodes that are in object mode.